### PR TITLE
redesign learner page

### LINKED ISSUE #3355: Fix display of Course Grades on the user profile (admin view)
The user profile, when viewed by an admin (faculty or staff) resues the Course component from the user's course card. The layout should be essentially the same, except that we will omit certain things that are not relevant to the admin user—for example, the links to "Contact Course Team" and "View Course on edX". We will also omit the status message box (the gray box) as it is geared toward the user, with call-to-action buttons etc.

The layout of this box on the user profile is currently pretty messed up looking, so we will need to fix the styles here to be consistent with the Course card layout.

Currently it looks like this:

image

So we should do this:
- Use the same element from the course card, with identical styling and layout.
- Omit the the links to "Contact Course Team" and "View Course on edX". 
- Omit the status message box (the gray box) element.
- In place of the status box, we will display small statuses just as text, where the "Contact Course Team" and "View Course on edX" links usually go. These will display the statuses in a simpler wording, and not include "You are...". For example, instead of "You are auditing" or "You missed the payment deadline", we will just say "Auditing" and "Missed payment deadline" and "Paid for course but not currently enrolled." We do not need to show other statuses.

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #3355 

#### What's this PR do?

this is a pretty simple change the adapts the `CourseListCard` component for the Learner page by adding a `showStaffView` prop that hides certain pieces of information, changes the title, etc.

#### How should this be manually tested?

take a look at the learner page, and make sure you see some details about programs you are enrolled in. you will need to be a staff user for the card to show up.

### CODE REVIEW

**Comment by noisecapella** [comment]
I'm getting an error:

screenshot from 2017-07-06 15-10-25

I think it happens when `currentProgramEnrollment` is not set, which is true briefly when a user is first accessing the site. Can you add a check in `FinancialAidCalculator` in case it's null?

**Comment by noisecapella** [comment]
Some of the states don't have messages like they do in the dashboard. For example this is for a failed course:

screenshot from 2017-07-06 16-25-54

**Comment by noisecapella** [comment]
Nevermind, looks like this is intentional per the issue

**Comment by noisecapella** [comment]
Here's the screenshots of the various dashboard states:

[dashboard_states_pr_3373.zip](https://github.com/mitodl/micromasters/files/1129217/dashboard_states_pr_3373.zip)

**Comment by roberthouse54** [comment]
This needs to show a status text under the grades, made up of two parts, separated by a dash. The first part shows whether the course is currently in progress, or not yet started, or ended. The second part shows whether the user is Auditing (with payment deadline date), Paid, Audited (if the course has ended), or Paid but not enrolled (which is an error state). See mock below.

image

File: static/js/components/Learner.js [hunk 1]
```diff
@@ -62,12 +63,21 @@ export default class Learner extends React.Component {
 
     if (! R.isEmpty(dashboard)) {
       return dashboard.programs.map(program => (
-        <StaffLearnerInfoCard
-          program={program}
-          key={program.title}
-          setShowGradeDetailDialog={setShowGradeDetailDialog}
-          dialogVisibility={ui.dialogVisibility}
-        />
+        <div key={program.title}>
```
Comment (noisecapella) [inline static/js/components/Learner.js]: Can we use `program.id` here instead? `title` isn't technically guaranteed to be unique

**Comment by aliceriot** [comment]
@noisecapella I believe I addressed all of what needs to happen (the new small course message display). It should now have what's described in the issue.

**Comment by noisecapella** [comment]
Here's the screenshot we were looking at:

dashboard:
dashboard_state_004_set_to_passed

learner:
dashboard_state_004_set_to_passed

**Comment by noisecapella** [comment]
For `014_set_past_run_to_passed` there is an auditing upgrade message in the dashboard

dashboard_state_014_set_past_run_to_passed


but there's no auditing message in the learner page:

dashboard_state_014_set_past_run_to_passed

**Comment by noisecapella** [comment]
If the user passes the edX course but doesn't pass the exam, they see this message on the dashboard:

dashboard_state_018_create_exams_edx_ _exam_ _with_new_offering

but they only see 'Passed' on the learner page:

dashboard_state_018_create_exams_edx_ _exam_ _with_new_offering

**Comment by aliceriot** [comment]
for the first case (the auditing message) do you have the JSON for the API response? I'm not exactly sure how to address that case.

**Comment by aliceriot** [comment]
@noisecapella bump

**Comment by pdpinch** [comment]
What's the status of this?

**Comment by aliceriot** [comment]
still in review

**Comment by noisecapella** [comment]
Sorry for the delay, I'll go through the screenshot differences and paste the ones with the JSON files in separate comments

**Comment by noisecapella** [comment]
`004_set_to_passed` has a dash on the dashboard page without any text. On master it isn't there:

this branch:
dashboard_state_004_set_to_passed

[dashboard_state_004_set_to_passed.zip](https://github.com/mitodl/micromasters/files/1180762/dashboard_state_004_set_to_passed.zip)

master:
[dashboard_state_004_set_to_passed-master.zip](https://github.com/mitodl/micromasters/files/1180766/dashboard_state_004_set_to_passed-master.zip)

**Comment by noisecapella** [comment]
There are similar changes for other dashboard screenshots. Running through the learner page screenshots now...

**Comment by noisecapella** [comment]
I'm still seeing this error on occasion when running through the screenshots: https://github.com/mitodl/micromasters/pull/3373#issuecomment-313491867

**Comment by noisecapella** [comment]
Here's all of the screenshots I took with the API JSON data, for both the dashboard page and the learner page. I'll comment again with the differences I noticed

[screenshots.zip](https://github.com/mitodl/micromasters/files/1180847/screenshots.zip)

**Comment by noisecapella** [comment]
- In the dashboard page for 001_set_to_enrolled there's a dash after `Course in progress -` which isn't there in master.
 - In the learner page for 004_set_to_passed there is a `- Passed` without anything before the dash. In the dashboard page there is a dash where there wasn't anything before in master
 - In the learner page for 014_set_past_run_to_passed it used to say `Auditing` in master but in this PR it doesn't say anything
 - In the learner page for 016 it says `- Passed` while also having the icon for `Passed` (not sure if this matters). On master it says `Paid` instead
 - In the learner page for 024_failed_with_prev_passed_run you have a passed icon but the text says `- Paid, did not pass`
 - In 037_missed_payment_can_reenroll the dashboard shows a 'Pay now' button but the learner page says `Course in progress - Missed payment deadline`

**Comment by aliceriot** [comment]
@noisecapella I can't seem to reproduce that error (with the financial aid calculator), do you have any ideas about what is causing it?

**Comment by aliceriot** [comment]
ok, going through and addressing the screenshots:

- 001_set_to_enrolled: the draft isn't showing up locally for me:

no-dash

 - 004_set_to_passed: there was a `- Passed` without anything before the dash, just fixed it:

passssseed

 - In the learner page for 014_set_past_run_to_passed it used to say `Auditing` in master but in this PR it doesn't say anything

I looked at the JSON, the issue for this one is that the `end_date` for the course is in the past. we only want to show the 'Auditing' message if the user is currently auditing an ongoing course, I think.

     
 - In the learner page for 016 it says `- Passed` while also having the icon for `Passed` (not sure if this matters). On master it says `Paid` instead

I think this is the intended behavior, the designs don't show any changes to the grades component (what draws the icon) for the learner page.

 - In the learner page for 024_failed_with_prev_passed_run you have a passed icon but the text says `- Paid, did not pass`

Just fixed this, just a little logical error. The function that returns 'Paid, did not pass' was only checking if the most recent course run was passed, where it should be checking all course runs.

 - In 037_missed_payment_can_reenroll the dashboard shows a 'Pay now' button but the learner page says `Course in progress - Missed payment deadline`

it looks like that's the existing behavior on the dashboard, but that doesn't seem right (why would we should 'pay now' if you missed the deadline?). Do you think we should address that here? (or am I misunderstanding something)

**Comment by noisecapella** [comment]
@aliceriot Regarding the financial aid calculator exception it happened during the `snapshot_dashboard_states` command. I'll run it again and let you know if it's still a problem

**Comment by noisecapella** [comment]
I filed a separate PR #3458 for the null reference thing

**Comment by noisecapella** [comment]
On the dashboard page in this branch I'm seeing `Course in progress -`:
dashboard_state_000_set_to_enrolled

**Comment by noisecapella** [comment]
Other than that everything looks great. I filed https://github.com/mitodl/micromasters/issues/3406 for the dashboard error, probably out of scope for here

**Comment by aliceriot** [comment]
ok, I think I fixed it!

**Comment by aliceriot** [comment]
alright, here we go

**Pull request merged**